src/backend/columnar/columnar_customscan.c

```diff
@@ -825,8 +1368,40 @@ ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 
 	if (scanDesc != NULL)
 	{
-		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
-		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
-							   chunkGroupsFiltered, es);
+		CustomScan *cscan = castNode(CustomScan, node->ss.ps.plan);
+		List *chunkGroupFilter = cscan->custom_exprs;
+
+		if (chunkGroupFilter != NIL)
+		{
+			Expr *conjunction;
+
+			if (list_length(chunkGroupFilter) == 1)
+			{
+				conjunction = (Expr *) linitial(chunkGroupFilter);
+			}
+			else
+			{
+				conjunction = make_andclause(chunkGroupFilter);
+			}
+
+#if PG_VERSION_NUM >= 130000
+			List *context = set_deparse_context_plan(es->deparse_cxt,
+													 &cscan->scan.plan,
+													 ancestors);
+#else
+			PlanState *planstate = &node->ss.ps;
+			List *context = set_deparse_context_planstate(es->deparse_cxt,
+														  (Node *) planstate,
+														  ancestors);
+#endif
+
+			char *str = deparse_expression((Node *) conjunction,
+										   context, false, false);
```

Comment:
what about putting this whole block into a function like `ColumnarScanChunkGroupFilterExprStr` ?

Reply:
The function isn't too big -- explain is mostly just doing that anyway. I'll wait until the projection explain patch is in, and then we can refactor them together and hopefully reuse some code.

Reply:
Yeah it's not very big but I find it useful to use the functions as building blocks for the sake of readability, but we can do that later as you suggested.